src/main.rs

```diff
@@ -3,70 +3,85 @@ extern crate structopt;
 #[macro_use]
 extern crate log;
 
+use lorri::locate_file;
+use lorri::NixFile;
+
 use lorri::cli::{Arguments, Command};
-use lorri::ops::{build, direnv, info, init, shell, upgrade, watch, ExitError, OpResult};
-use lorri::project::{Project, ProjectLoadError};
+use lorri::ops::{
+    build, daemon, direnv, info, init, ping, shell, upgrade, watch, ExitError, OpResult,
+};
+use lorri::project::Project;
 use std::env;
 use structopt::StructOpt;
 
 const TRIVIAL_SHELL_SRC: &str = include_str!("./trivial-shell.nix");
 const DEFAULT_ENVRC: &str = "eval \"$(lorri direnv)\"";
 
 fn main() {
+    let exit = |result: OpResult| match result {
+        Err(err) => {
+            eprintln!("{}", err.message());
+            std::process::exit(err.exitcode());
+        }
+        Ok(Some(msg)) => {
+            println!("{}", msg);
+            std::process::exit(0);
+        }
+        Ok(None) => {
+            std::process::exit(0);
+        }
+    };
+
     let opts = Arguments::from_args();
 
     lorri::logging::init_with_default_log_level(opts.verbosity);
     debug!("Input options: {:?}", opts);
 
-    let project = Project::from_cwd();
-
-    let result: OpResult = match (opts.command, project) {
-        (Command::Info, Ok(project)) => info::main(&project),
-
-        (Command::Build, Ok(project)) => build::main(&project),
+    match lorri::ops::get_paths() {
```

Comment:
Let's undo this part. Many commands present and future may not require a local project in the current directory, which is something the previous version of this supported explicitly. This will fix my first issue with the daemon:

```
[grahamc@Petunia:~]$ lorri daemon
There is no `shell.nix` in the current directory (/home/grahamc)
You can use the following minimal `shell.nix` to get started:

let
  pkgs = import <nixpkgs> {};
in
  pkgs.mkShell {
    buildInputs = [
        pkgs.hello
    ];
  }
```

Reply:
We talked about this: having the shell.nix check explicitly means we can give a nicer more specific error message, so +1 on what is here now.